app/models/services/service_key.rb

```diff
@@ -79,8 +81,44 @@ def logger
       @logger ||= Steno.logger('cc.models.service_key')
     end
 
+    def last_operation
+      service_key_operation
+    end
+
+    def terminal_state?
+      !service_key_operation || (%w(succeeded failed).include? service_key_operation.state)
+    end
+
+    # TODO: can these methods be a mixin somehow
     def operation_in_progress?
-      false
+      !!service_key_operation && service_key_operation.state == 'in progress'
+    end
+
+    def required_parameters
+      nil
+    end
+
+    def save_with_attributes_and_new_operation(attributes, operation)
+      save_with_new_operation(operation, attributes: attributes)
```

Comment:
I think `save_with_attributes_and_new_operation ` is the standard method that routes and service bindings are using from the actions. Might be worth having only that one and inlining `save_with_new_operation` (which is inherited from the initial implementation of service bindings)

Reply:
we are currently using both for some reason throughout the bindings actors/actions. I'll add a chore to take a look after this.